Gems/AtomTressFX/Code/Components/HairComponentController.cpp

```diff
@@ -0,0 +1,373 @@
+
+#include <Integration/Components/ActorComponent.h>
+#include <EMotionFX/Source/TransformData.h>
+#include <EMotionFX/Source/ActorInstance.h>
+#include <EMotionFX/Source/Node.h>
+
+// Hair Specific
+#include <TressFX/TressFXAsset.h>
+#include <TressFX/TressFXSettings.h>
+
+#include <Rendering/HairFeatureProcessor.h>
+#include <Components/HairComponentController.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        namespace Hair
+        {
+            HairComponentController::~HairComponentController()
+            {
+                RemoveHairObject();
+            }
+
+            void HairComponentController::Reflect(ReflectContext* context)
+            {
+                HairComponentConfig::Reflect(context);
+
+                if (auto* serializeContext = azrtti_cast<SerializeContext*>(context))
+                {
+                    serializeContext->Class<HairComponentController>()
+                        ->Version(2)
+                        ->Field("Configuration", &HairComponentController::m_configuration)
+                        ;
+                }
+
+                if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+                {
+                    behaviorContext->EBus<HairRequestsBus>("HairRequestsBus")
+                        ->Attribute(AZ::Script::Attributes::Module, "render")
+                        ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+
+                        // Insert auto-gen behavior context here...
+                        ;
+                }
+            }
+
+            void HairComponentController::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+            {
+                provided.push_back(AZ_CRC_CE("HairService"));
+            }
+
+            void HairComponentController::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+            {
+                incompatible.push_back(AZ_CRC_CE("HairService"));
+            }
+
+            void HairComponentController::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+            {
+                // Dependency in the Actor due to the need to get the bone / join matrices
+                required.push_back(AZ_CRC_CE("EMotionFXActorService"));
+            }
+
+            HairComponentController::HairComponentController(const HairComponentConfig& config)
+                : m_configuration(config)
+            {
+            }
+
+
+            void HairComponentController::Activate(EntityId entityId)
+            {
+                m_entityId = entityId;
+
+                m_featureProcessor = RPI::Scene::GetFeatureProcessorForEntity<Hair::HairFeatureProcessor>(m_entityId);
+                if (m_featureProcessor)
+                {
+                    m_featureProcessor->SetHairGlobalSettings(m_configuration.m_hairGlobalSettings);
```

Comment:
So this might be a temporary solution but basically the global settings is only used for shader options.
These options are being used by all hair components and the two approaches were:
1. Create a stand alone component and editor representation for all Hair in general and connect to all Hair objects / components.
2. Create this as part of the hair representation with the assumption that changing it is affecting and reflecting all.

Robin at the time chose the second option and there are benefit and downsides:
Benefits:
- Simpler 
- The implementation was designed to be a stand alone class that can be used differently in the future 
- Intuitive - immediate representation to the user at the same hair control location
- Simpler user creation and usage - no need of extra creation steps or looking for these options when wanting to change

Downside:
- Being global conceptually means apart from the per-object, matter of choice really
- Needs to reflect into all Hair objects - this is done exactly the same as would be with global stand alone component, so the design is the same.
- No stand alone editor component - not sure this is a downside though.